### ISSUE TITLE
Content recommendations (like Next Steps, Resume) do not update when returning to the Learn page

### ISSUE THREAD

**Issue by jeepurs** [learningequality/kolibri#2414]
## Summary

 After partially interacting with a content item, the progress icon appears.  Upon returning to the Recommended page, 'Next steps' and 'Resume' are not updated with the recent interaction.  These items do appear, however, upon refreshing the Learn page within the browser.

Alternative title: There is no invalidation of the client side cache of recommendations.

## System information

*Please specify the Kolibri version you were using and your operating system!*

 - Version: 0.6 beta
 - Operating system: Online demo server
 - Browser: Chrome

## How to reproduce

1. Interact with content (without completion)
2. Return to the Learn page by clicking on 'Recommended' in the top left corner of the screen
3. Expect to see the item under "Resume" or see irrelevant "Next steps", but they aren't there yet

## Screenshots

Third item in the first row of content thumbnails shows partial progress, but isn't included under 'Resume':
![error](https://user-images.githubusercontent.com/5150483/31414754-5d0a0438-add4-11e7-85db-766f572c30af.png)

## Real-life consequences (anything community should be aware of, for instance how it affects your deployment)

Self-paced progress is confusing for the learner

**Comment by jonboiser** [comment]
@jeepurs In the title of the issue, did you mean the recommendations _do not_ update?

So the expectation is if you partially interact with a Content Item A, then return to the Learn home page, then:

1. Item A will appear in the "Resume" row
1. Any "Next steps" items associated with Item A will appear in the "Next Steps" row

**Comment by jeepurs** [comment]
@jonboiser thanks for catching that glaring mistake.  I updated the title.  Yes, the expectations you've written are accurate.

**Comment by rtibbles** [comment]
Note that this is not as simple as invalidating the recommendations cache once content has been interacted with, as we have deliberately maintained the heavy caching on the recommendation endpoints as they are intensive.

The desired solution here is for us to have more aggressive server side caching of recommendations.

My suggestion is that Kolibri generated personalized recommendations are put into a separate Model that these endpoints now return, and which are updated by, for example, engaging with content.

**Comment by jonboiser** [comment]
I just revisited this on the kolibri-beta server, and it still seems to be happening. I watched a video, got it to an "in-progress" state, but did not see it appear in the "resume" row even after refreshing.

Looking at the code, is this sorting correct? 

https://github.com/learningequality/kolibri/blob/6443d53eea37bcd8d33c42c592ffb93ab2074f22/kolibri/core/content/api.py#L707-L708

Intuitively, I would see the video I just watched at the top of my results, but only if we sorted the end timestamp in _descending_ order. Not sure if that's the default behavior of this `order_by("end_timestamp")` call.
